internal/js/summary.js

```diff
@@ -235,17 +237,17 @@ class ReportBuilder {
 	 * Adds groups sections to the report.
 	 *
 	 * @param {Record<string, ReportGroup>} groups - The groups to add to the report.
+	 * @param {string[]} groupsOrder - Group names sorted with the order to be displayed in the summary. Typically same as in code.
 	 * @returns {ReportBuilder}
 	 */
-	addGroups(groups) {
+	addGroups(groups, groupsOrder) {
 		if (!groups) return this;
 
-		Object.entries(groups)
-			.sort(([a], [b]) => a.localeCompare(b))
-			.forEach(([groupName, groupData]) => {
+		groupsOrder
+			.forEach((groupName) => {
```

Comment:
JS maps are [ordered](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map#description) in insertion order. 
This was used in the previous summary to not have to require the rendering part to order them, but to order them before that. See 
https://github.com/grafana/k6/blob/0641c731e053718010a6a1bab6784209c3586439/internal/js/summary-wrapper.js#L18-L34
https://github.com/grafana/k6/blob/0641c731e053718010a6a1bab6784209c3586439/internal/js/summary-wrapper.js#L37-L61

It will be nice that instead of us changing this part of the code this just gets the `groups` argument in the correct order